### DISCUSSION TITLE
💲 Cost and usage reports for all AP accounts via Athena

### DISCUSSION THREAD

**Discussion by bagg3rs** [ministryofjustice/analytical-platform#5011] [Firebreak Summer 24]
### Idea Description

Following on from the last [successful firebreak](https://github.com/ministryofjustice/analytical-platform/discussions/2640) this looks to consolidate CURs or Cost and Usage Reports for our accounts into an S3 bucket and then present that with Athena/LakeFormation. This makes cost data available to many tools including QuickSight and Grafana.

### Why Should We Do This

We don't think about cost of services and by making it easier to view report we can look to lower costs and work on sustainability [TCOP 12](https://www.gov.uk/guidance/the-technology-code-of-practice#make-your-technology-sustainable)

### Definition of Done

- [ ] CURs from our accounts are exported to central account location (this could be via Root Account or billing account, chat with MP)
- [ ] Data crawled and registered in Glue

### What skills do we need?

AWS (Athena, Lakeformation), Terraform, if root account involved then lashings of charm

### Notes

See for example [here](https://community.amazonquicksight.com/t/visualize-amazon-quicksight-costs-using-aws-cur-and-cost-allocation-tags/25823)

**Comment by AntFMoJ** [comment]
Could we tack on getting carbon footprint/sustainability metrics from the Root or Billing account as well? 😄
